### ISSUE TITLE
Crash on Sidekiq web interface on each manual job operation "wrong number of arguments (given 1, expected 2)"

### ISSUE THREAD

**Issue by andreasunterhuber** [appsignal/appsignal-ruby#695]
Using Sidekiq in most recent version (6.1.3) and AppSignal in latest version (2.11.4) we get errors on web-ui of Sidekiq (Sidekiq::Web) we get errors on each job operation like `delete`, `re-enqueue`. 

## To Reproduce

- Using AppSignal 2.11.4 for Ruby gem, Ruby version 2.6.6
- Add a job that fails or starts in some hours for example
- open web-interface of Sidekiq and lick on re-enqueue
- Crash with the following  error `wrong number of arguments (given 1, expected 2)` in `appsignal (2.11.4) lib/appsignal/rack/rails_instrumentation.rb:32:in `call_with_appsignal_monitoring'`

**Comment by tombruijn** [comment]
Hi @andreasunterhuber, can you try upgrading to AppSignal for Ruby gem version 2.11.4 and let us know if still occurs then?

**Comment by andreasunterhuber** [comment]
@tombruijn as I mentioned in my post, we use already version `2.11.4`

**Comment by tombruijn** [comment]
Ah ok, you listed "2.6.6" later in your message, so I got confused 😅

**Comment by tombruijn** [comment]
Looks like the arguments have changed so we'll need to investigate what arguments are expected instead. Weird that it only happens for Sidekiq web UI in the Rails app.

I expect it's not a lot of work, but we'll need to plan some time for this. The team gets together tomorrow to discuss it.

**Comment by andreasunterhuber** [comment]
@tombruijn thanks for the feedback! Ok. would be great if the fix will arrive not so far in the future. It affects three productive environments on our side.

**Comment by tombruijn** [comment]
@andreasunterhuber I've tried to reproduce your issue in our test-apps using the `ruby/rails-sidekiq` app and haven't been able to reproduce the error in a basic Rails + Sidekiq + AppSignal app. Can you tell me more about what other frameworks your using in your Rails app? And how you mount the sidekiq web ui? What kind of web ui are you using, basic or the pro version?

**Comment by andreasunterhuber** [comment]
Hi @tombruijn thanks for trying to fix the issue. We use the standard Sidekiq with multiple instances and queues. 
* **Config 1**
   *  Sidekiq 5.2.7
   * Latest AppSignal 2.11.4
   * Additional Gems that could probably interfere
       * sidekiq-limit_fetch
       * sidekiq-scheduler
* **Config 2**
   *  Sidekiq 6.1.3
   * Latest AppSignal 2.11.4
   * Additional Gems that could probably interfere
       * sidekiq-limit_fetch
       * sidekiq-scheduler

Hope this info helps!

**Comment by andreasunterhuber** [comment]
@tombruijn we tried also with AppSignal version `2.11.5` without success.

**Comment by tombruijn** [comment]
@andreasunterhuber I am unable to get the error you're running into. I've tested this on this testing app with the additional sidekiq gems you listed: https://github.com/appsignal/test-setups/tree/sidekiq-issue-695

But so far no error when requeuing or deleting jobs from the sidekiq web UI. Can you compare the setup of your app to the one I linked? Or update it to create a reproducible state of the error you're running into.

**Comment by andreasunterhuber** [comment]
@tombruijn thanks a lot for your help. We will invest now some additional time to dig into it 👍 I will come back to you shortly.

**Comment by andreasunterhuber** [comment]
Dear @tombruijn it seems that this was caused by `rack-protection` GEM that was a decency from Sidekiq in the older days. It was still in our projects. The crash leads always to AppSignal, but AppSignal was not the cause. After cleaning up some code we removed it and it seems to work now 🎉 
We tested with two projects and it seems fine so far. I guess this is enough to close the issue.
Sorry for the effort on your side - but we are very happy that this is finally resolved!
